query.go

```diff
@@ -158,40 +162,78 @@ func getFrames(rows *sql.Rows, limit int64, converters []sqlutil.Converter, fill
 	frame.Meta.ExecutedQueryString = query.RawSQL
 	frame.Meta.PreferredVisualization = data.VisTypeGraph
 
-	if query.Format == FormatOptionTable {
-		frame.Meta.PreferredVisualization = data.VisTypeTable
-		return data.Frames{frame}, nil
+	count, err := frame.RowLen()
+	if err != nil {
+		return nil, err
+	}
+	if count == 0 {
+		return nil, ErrorNoResults
 	}
 
-	if query.Format == FormatOptionLogs {
+	switch query.Format {
```

Comment:
This is entirely a matter of taste, and I'm not sure I actually prefer it, but I might structure this a bit differently to make the exceptional case stand out a little more: https://github.com/grafana/sqlds/compare/iwysiu/96...njvrzm/96-semi_suggestion (I can't add this as a suggestion since the range crosses deleted lines, sigh)

Reply:
I think it makes sense. Done.